flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/declarative/MockExecutionGraph.java

```diff
@@ -69,15 +73,22 @@ public MockExecutionGraph() throws IOException {
                 1,
                 ExecutionGraph.class.getClassLoader(),
                 VoidBlobWriter.getInstance(),
-                PartitionReleaseStrategyFactoryLoader.loadPartitionReleaseStrategyFactory(
-                        new Configuration()),
+                getPartitionReleaseStrategyFactoryLoader(),
                 NettyShuffleMaster.INSTANCE,
                 NoOpJobMasterPartitionTracker.INSTANCE,
                 ScheduleMode.EAGER,
                 NoOpExecutionDeploymentListener.get(),
                 (execution, newState) -> {},
                 0L);
         this.setJsonPlan(""); // field must not be null for ArchivedExecutionGraph creation
+        this.attachJobGraph(Collections.emptyList()); // method must be called to initialize
+        // PartitionReleaseStrategy.
+    }
+
+    private static PartitionReleaseStrategy.Factory getPartitionReleaseStrategyFactoryLoader() {
+        Configuration conf = new Configuration();
+        conf.set(PARTITION_RELEASE_DURING_JOB_EXECUTION, false);
+        return PartitionReleaseStrategyFactoryLoader.loadPartitionReleaseStrategyFactory(conf);
```

Comment:
Hmm, I strongly suggest to add an interface for testing in the foreseeable future. I think we are seeing the consequences of mocking the `ExecutionGraph` by running into problems originating from setting it up wrongly.

Reply:
Yes, I agree. This is very fragile and annoying to maintain.